digest.go

```diff
@@ -84,17 +86,26 @@ func (a *DigestAuth) Purge(count int) {
  (or requires reauthentication).
 */
 func (a *DigestAuth) RequireAuth(w http.ResponseWriter, r *http.Request) {
+	a.mutex.RLock()
```

Comment:
Which data structure(s) is exactly protected by the `mutex`?

Reply:
the `mutex` protect only `clients`

Reply:
I guess the `Purge` method is called then from other places as well? If not the locking could be re-organized so that only one write lock is acquired instead of a read lock and two write locks.

If it is called by other places and it definitely needs it's own write lock than the current implementation looks like all we can do. As there is no unit-test here, did you verify that it works properly to acquire a read lock and in the same process the write lock as well?

Reply:
After some investigation, it is not used from other place. So I re-organized locks